### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by gitforhongliming**
optimize fs (error,remove ioctrl,open)，fix readdir length error

### CODE REVIEW

File: components/fs/fatfs/los_fatfs.c [hunk 1]
```diff
@@ -356,7 +356,13 @@ static off_t fatfs_op_lseek (struct file *file, off_t off, int whence)
 
     FRESULT res = f_lseek(fp, off);
     if (res == FR_OK)
-        return off;
+    {
+    	if(off < 0)
+    	{
+    		ret_to_errno(FR_INVALID_PARAMETER);
```
Comment (jimscen-wang) [inline components/fs/fatfs/los_fatfs.c]: return ret_to_errno()?

File: components/fs/fatfs/los_fatfs.c [hunk 2]
```diff
@@ -450,7 +456,7 @@ static int fatfs_op_readdir (struct dir *dir, struct dirent *dent)
         return ret_to_errno(res);
     }
 
-    len = MIN(sizeof(e.fname), LOS_MAX_FILE_NAME_LEN) - 1;
+    len = MIN(sizeof(e.fname), LOS_MAX_DIR_NAME_LEN) - 1;
```
Comment (jimscen-wang) [inline components/fs/fatfs/los_fatfs.c]: len = MIN(sizeof(e.fname), LOS_MAX_DIR_NAME_LEN + 1) - 1 ?

File: include/fs/los_vfs.h [hunk 3]
```diff
@@ -40,6 +40,7 @@
 #include <los_typedef.h>
 #include "fs/sys/stat.h"
 
+#define LOS_MAX_DIR_NAME_LEN                 255
```
Comment (jimscen-wang) [inline include/fs/los_vfs.h]: may include ffconf.h.
and #define LOS_MAX_DIR_NAME_LEN FF_LFN_BUF ?

File: components/fs/fatfs/los_fatfs.c (hunk 1 above)
Comment (gitforhongliming) [reply components/fs/fatfs/los_fatfs.c]: yes，this only set errno as off < 0 the result of f_lseek(fp, off) still return success. The value of result (fatfs_op_lseek()) is through "return off"

File: include/fs/los_vfs.h (hunk 3 above)
Comment (gitforhongliming) [reply include/fs/los_vfs.h]: not need .as los_vfs.h is used for all file system .Not need include fat.If do this ,I think it is not reasonalbe.

File: components/fs/fatfs/los_fatfs.c (hunk 2 above)
Comment (gitforhongliming) [reply components/fs/fatfs/los_fatfs.c]: ok,should change LOS_MAX_FILE_NAME_LEN to LOS_MAX_DIR_NAME_LEN+1

File: include/fs/los_vfs.h (hunk 3 above)
Comment (jimscen-wang) [reply include/fs/los_vfs.h]: but I think the LOS_MAX_DIR_NAME_LEN and FF_LFN_BUF should define in one place, then it will keep the same all the time.

**Pull request merged**